pkg/resourcemonitor/noderesourcesaggregator.go

```diff
@@ -0,0 +1,243 @@
+	perNUMACapacity map[int]map[v1.ResourceName]int64
+	// we should use a different map for each resource type: resourceName -> resourceID -> nodeID
+	resourceID2NUMAID map[string]map[string]int
+	cpuInfo           *cpus.CPUs
+	numaInfo          *numa.Nodes
+	numaDists         *distances.Distances
+}
+
+type ResourceData struct {
+	allocatable int64
+	capacity    int64
+}
+
+func NewResourcesAggregator(sysfsPath string, podResourceClient podresourcesapi.PodResourcesListerClient) (ResourcesAggregator, error) {
+	var err error
+
+	cpuInfo, err := cpus.NewCPUs(sysfsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	nodes, err := numa.NewNodesFromSysFS(sysfsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	dists, err := distances.NewDistancesFromSysfs(sysfsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultPodResourcesTimeout)
+	defer cancel()
+
+	//Pod Resource API client
+	resp, err := podResourceClient.GetAvailableResources(ctx, &podresourcesapi.AvailableResourcesRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("Can't receive response: %v.Get(_) = _, %v", podResourceClient, err)
+	}
+
+	allDevs := GetContainerDevicesFromAllocatableResources(resp, cpuInfo)
+
+	nodeResourceInstance := nodeResources{
+		cpuInfo:           cpuInfo,
+		numaInfo:          &nodes,
+		numaDists:         dists,
+		resourceID2NUMAID: makeResourceMap(len(nodes.Online), allDevs),
+		perNUMACapacity:   MakeNodeCapacity(allDevs),
+	}
+
+	return &nodeResourceInstance, nil
+}
+
+func (nodeResourceData *nodeResources) Aggregate(podResData []PodResources) map[string]*v1alpha1.Zone {
+	perNuma := make(map[int]map[v1.ResourceName]*ResourceData)
+	for nodeNum, nodeRes := range nodeResourceData.perNUMACapacity {
+		perNuma[nodeNum] = make(map[v1.ResourceName]*ResourceData)
+		for resName, resCap := range nodeRes {
+			perNuma[nodeNum][resName] = &ResourceData{
+				capacity:    resCap,
+				allocatable: resCap,
+			}
+		}
+	}
+
+	for _, podRes := range podResData {
+		for _, contRes := range podRes.Containers {
+			for _, res := range contRes.Resources {
+				nodeResourceData.updateAllocatable(perNuma, res)
+			}
+		}
+	}
+
+	zones := make(map[string]*v1alpha1.Zone)
+
+	for nodeNum, resList := range perNuma {
+		zone := &v1alpha1.Zone{
+			Type:      "Node",
+			Resources: make(v1alpha1.ResourceInfoMap),
+		}
+
+		costs, err := makeCostsPerNumaNode(nodeResourceData.numaInfo, nodeResourceData.numaDists, nodeNum)
+		if err != nil {
+			// TODO log and move forward
+		} else {
+			zone.Costs = costs
+		}
+
+		for name, resData := range resList {
+			allocatableQty := *resource.NewQuantity(resData.allocatable, resource.DecimalSI)
+			capacityQty := *resource.NewQuantity(resData.capacity, resource.DecimalSI)
+			zone.Resources[name.String()] = v1alpha1.ResourceInfo{
+				Allocatable: allocatableQty.String(),
+				Capacity:    capacityQty.String(),
+			}
+		}
+		zones[makeZoneName(nodeNum)] = zone
+	}
+	return zones
+}
+
```

Comment:
Lets add a comment providing a brief description of what this function does

Reply:
done